internal/resources/cluster/nodepools/data_source_node_pool.go

```diff
@@ -13,21 +13,40 @@ import (
 	"github.com/pkg/errors"
 
 	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/authctx"
+	clienterrors "github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/errors"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
 	nodepoolsmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/cluster/nodepool"
 	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
 )
 
 func DataSourceClusterNodePool() *schema.Resource {
 	return &schema.Resource{
-		ReadContext: dataSourceClusterNodePoolRead,
-		Schema:      nodePoolSchema,
+		ReadContext: func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+			return dataSourceClusterNodePoolRead(helper.NewCallerContext(ctx).GetContextWithCaller(helper.DataRead), d, m)
+		},
+		Schema: nodePoolSchema,
 	}
 }
 
-func dataSourceClusterNodePoolRead(_ context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+func dataSourceClusterNodePoolRead(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
 	config := m.(authctx.TanzuContext)
 
-	resp, err := config.TMCConnection.NodePoolResourceService.ManageV1alpha1ClusterNodePoolResourceServiceGet(constructFullName(d))
+	fn := constructFullName(d)
+	resp, err := config.TMCConnection.NodePoolResourceService.ManageV1alpha1ClusterNodePoolResourceServiceGet(fn)
+
+	if err != nil {
+		if clienterrors.IsNotFoundError(err) {
+			if helper.NewCallerContext(ctx).IsDataRead() {
```

Comment:
I think, we need to just flip the condition because we have a return with error below

Reply:
@shreyassreenivas I don't think we need to flip it based on similar check for cluster resource data read - [data_source_cluster.go#L55](https://github.com/vmware/terraform-provider-tanzu-mission-control/blob/143c1e710f639065d1924cfe57d479be64720867/internal/resources/cluster/data_source_cluster.go#L55)

Reply:
This error condition will be handled below, so flip the condition